app/App.tsx

```diff
@@ -23,6 +28,24 @@ function App() {
     return null;
   }
 
+  if (fileInfo.size === 0) {
+    // e.g. when comparing git changes on an untracked file - https://github.com/silx-kit/vscode-h5web/issues/22
+    return <p>File does not exist</p>;
+  }
+
+  if (fileInfo.size >= MAX_SIZE_IN_BYTES) {
+    return (
+      <StandaloneViewer
+        customMessage={
+          <p>
+            File is too large to be opened from the explorer (max 2 GB). Please
+            browse for it from here:
+          </p>
+        }
```

Comment:
Perhaps a premature abstraction, but it makes `StandaloneViewer` usable in a webview on its own if need be.